lms/djangoapps/grades/tasks.py

```diff
@@ -0,0 +1,54 @@
+"""
+This module contains tasks for asynchronous execution of grade updates.
+"""
+
+from celery import task
+from django.contrib.auth.models import User
+
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from lms.djangoapps.courseware.courses import get_course_by_id
+from opaque_keys.edx.keys import UsageKey
+from opaque_keys.edx.locator import CourseLocator
+from openedx.core.djangoapps.content.block_structure.api import get_course_in_cache
+
+from .config.models import PersistentGradesEnabledFlag
+from .transformer import GradesTransformer
+from .new.subsection_grade import SubsectionGradeFactory
+
+
+@task()
+def recalculate_subsection_grade(user_id, course_id, usage_id):
```

Comment:
Yeah, I think this is the way to go for now. The subsection update task will fire a `COURSE_GRADE_NEEDS_UPDATE` signal (name TBD), and perform all of that updating in a listener inside the same process. If it turns out that there is value in having that listener be a separate task, we can split it out at that point.